[Leong See Leng] iP

seelengxd opened the pull request:
# Duke
 ![duke](https://github.com/seelengxd/ip/blob/master/src/main/resources/images/duke.png?raw=true)
- duke does many things
1. todos
2. events
3. deadlines

```java
System.out.println("Duke is written in Java :(");
```

duke can record ur task lists
this is a markdown task list :clown_face:

- [ ] todo
- [ ] important very important event /at 2022-05-06

> i have **cool** [duke icons](https://github.com/seelengxd/ip/tree/master/src/main/resources/images)

`inline code ok`
some text formatting: **bold**, *italic*, ~~stikethrough~~ etc.

Commit:
Add Gradle support

seelengxd committed:
Level-1: add greet, echo, exit

Also changed colour of Duke logo to cyan to make it look nice

seelengxd committed:
Refactor to use switch statement in main

seelengxd committed:
Level-2: add, list

seelengxd committed:
Level-3: mark as done

Task has been abstracted out to its own class.

seelengxd committed:
A-Inheritance: add Deadline, Event, Todo classes

All 3 classes inherit from Task and Task's description
access modifier is now default so its subclasses can access it.

seelengxd committed:
fix Deadline and Event display and inheritance

Fix careless mistakes from previous commit. Now displays done flags and
type flags correctly.

seelengxd committed:
Level-4: add todo, deadlines, event commands

seelengxd committed:
Level-5: handle exceptions

Created a class DukeException to handle user input errors

seelengxd committed:
Handle task index related exceptions

seelengxd committed:
Level-6: delete task

seelengxd committed:
Add missing print for list command

seelengxd committed:
A-TextUiTesting: add test input and output files

seelengxd committed:
A-Enums: refactor addTask to use taskType enum

seelengxd committed:
Level-7: Add save feature

seelengxd committed:
Level-8: add dates

seelengxd committed:
Merge branch 'branch-Level-7'

seelengxd committed:
Merge branch 'branch-Level-8'

Also handles LocalDate for file save and load

seelengxd committed:
A-MoreOOP: Refactor Duke to group related code into more classes

Classes Extracted: Command classes, Ui, Parser, Storage, TaskList

seelengxd committed:
A-Packages: separate classes into duke and duke.command

seelengxd committed:
Merge remote-tracking branch 'upstream/add-gradle-support'

seelengxd committed:
Fix gradle package

My repo does not use "seedu"

seelengxd committed:
Add getMessage to DukeException

seelengxd committed:
A-JUnit: add junit tests

seelengxd committed:
A-Javadoc: add javadoc to class headers and public methods

seelengxd committed:
A-Checkstyle: Set up checkstyle

To make it easier to follow the coding standard.

seelengxd committed:
A-CodingStandard: Fix checkstyle violations

seelengxd committed:
Level-9: add find command

seelengxd committed:
Merge branch 'branch-A-JavaDoc'

seelengxd committed:
Merge branch 'branch-A-CodingStandard'

seelengxd committed:
Merge branch 'branch-Level-9'

Also fixes more FindCommand related style issues.

seelengxd committed:
Refactor: fix javadoc header comments

First sentence should start with a word with 's' and end with a '.'

seelengxd committed:
Refactor: rename taskList to tasks

Fix java coding standard violation: `Plural form should be used on names
representing a collection of objects.`

seelengxd committed:
Add JavaFX to build.gradle

seelengxd committed:
Hooks up JavaFX to dummy getResponse command

seelengxd committed:
Level-10: integrate JavaFX GUI with duke commands

- Refactored commands to return string responses instead of
printing.
- New Message class to send flags to GUI.

seelengxd committed:
Merge branch 'branch-Level-10'

TanWenCong commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,80 @@
+     * @param rep String of task representation
+     * @return Nothing.
+     */
+    public static Task fromFileRepresentation(String rep) {
```
Since child classes of Task will override this method, perhaps an abstract method might be preferred?

TanWenCong commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,67 @@
+    private LocalDate by;
+
+    /**
+     * Creates a new Deadline.
```
Very minor thing, but following CS2103T's Java coding standard, 
> Empty line between description and parameter section 

they would require an empty line here. However, if the checkstyle script (in week 4's increment A-checkstyle) did not raise this up, I believe it would be fine to leave it as it is.

TanWenCong commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,138 @@
+     * @param date
+     * @return ArrayList of tasks on specified date
+     */
+    public ArrayList<Task> getTasksOn(LocalDate date) {
```
I like how you chained the functions together to produce a clean one-liner!

TanWenCong commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,80 @@
+/**
+ * Abstract class that all Tasks inherit from.
+ */
+public abstract class Task {
```
Great use of abstract class to reduce code duplication!

TanWenCong reviewed (commented):
Awesome code quality overall, LGTM!

seelengxd committed:
Refactor: add blank line in javadoc between description and params

There are several javadocs for methods without a blank line between the description and params.

They are fixed in this commit to follow the style guide.

Also removed unused imports.

seelengxd committed:
A-Assertions: add assertions to duke

No assertions in code.

Assertions are useful in helping verify assumptions of variable state in
the program and detecting bugs.

Let's add some assertions.

seelengxd committed:
Merge pull request #1 from seelengxd/branch-A-Assertions

A-Assertions: added assertions to duke

samuelcheongws commented on src/main/java/duke/Event.java:
```diff
@@ -0,0 +1,72 @@
+    /**
+     * Returns file representation of Event.
+     *
+     * @return "E | {1 if done else 0} | {description} | {at}"
```
I like the clarity of the return statements.

samuelcheongws commented on src/main/java/duke/Message.java:
```diff
@@ -0,0 +1,36 @@
+ * Represents the message with appropriate flags needed by the display.
+ */
+public class Message {
+    private boolean isExit;
```
You could consider using shouldExit. However, I do think isExit is also a very reasonable boolean name.

samuelcheongws commented on src/main/java/duke/command/FindCommand.java:
```diff
@@ -0,0 +1,46 @@
+     */
+    @Override
+    public String execute(TaskList tasks, Ui ui, Storage storage) throws DukeException {
+        ArrayList<Task> filteredTasks = tasks.searchTasks(searchQuery);
```
You could consider using a more spcific list here such as TaskList. :)

samuelcheongws reviewed (commented):
LGTM. Overall I see very consistent effort put to maintain good coding standard. Your code is easy to read. There is no apparent coding standard violations. I gave some suggestions for your consideration!

seelengxd committed:
Refactor: Extract filter logic to reusable method

Filter logic is duplicated in `searchTasks` and `getTasksOn`.

This decreases readability and makes it difficult to add more filters in
the future.

Let's extract out filter logic to separate method `filterTasks`.

seelengxd committed:
Refactor: constantize DukeException error messages

Error messages are being passed into DukeExceptions as magic strings.

This reduces readability and will cause code duplication when throwing
similar errors in the future.

Let's refactor by moving the error messages to individual constants.

seelengxd committed:
Merge pull request #2 from seelengxd/branch-A-CodeQuality

A-CodeQuality: reduce code duplication and magic numbers

seelengxd committed:
A-CI: add gradle.yml

- Replaced text-io redirection with running gradle test cases.

seelengxd committed:
Refactor: remove trailing whitespace in Parser

It is causing CI to fail...

seelengxd committed:
C-MassOps: support deleting multiple tasks at one go

seelengxd committed:
Merge pull request #3 from seelengxd/branch-C-MassOps

C-MassOps: support deleting multiple tasks at one go

seelengxd committed:
A-BetterGui: dark mode

GUI was in light mode, with ugly green and red colors. Images were
unnecessarily large.

Let's switch it to dark mode.

Color palette mostly from https://www.happyhues.co/palettes/4.

seelengxd committed:
Merge pull request #4 from seelengxd/branch-A-BetterGui

A-BetterGui: implement dark mode

seelengxd committed:
A-MoreErrorHanding: prevent invalid dates from crashing app

seelengxd committed:
Merge pull request #5 from seelengxd/branch-A-MoreErrorHandling

A-MoreErrorHandling: handle invalid dates

seelengxd committed:
A-UserGuide: add features with demo images to docs/README.md

seelengxd committed:
Merge pull request #6 from seelengxd/branch-A-UserGuide

A-UserGuide: add features with demo images to docs/README.md

seelengxd committed:
Refactor: make boolean sound like boolean

seelengxd committed:
Refactor: fix javadoc for TaskOnDateCommand

First word should start with 's'.

Related to #7

seelengxd committed:
Refactor: move classes to relevant packages

seelengxd committed:
Refactor: update Ui class docs

Ui class used to handle colors and lines, but now visual stuff are
almost all done by javafx.

seelengxd committed:
Fix: change xmlns attribute in xml file to 8.0.171

Replaced to tutorial's version since it seems to be causing warnings on
Windows (based on smoke test)

seelengxd committed:
Refactor: add quickstart to docs/README.md

Remind user to use terminal instead of double clicking on jar file.

seelengxd committed:
Merge pull request #8 from seelengxd/refactor

Improve code quality & fix javafx version warning reported in smoke test

seelengxd committed:
Refactor: make ui a static class

It was unnecessary to make ui an instance variable to be passed in
commands, and some commands do not even use it!

Refactor to static class that commands that need it can import.

seelengxd committed:
Fix: disable marking an already done task

Also disables unmarking a not done task.

seelengxd committed:
Test: add tests for command classes

seelengxd committed:
Feat: add equals method to task classes

Added equals method for easier writing of tests.

seelengxd committed:
Test: add tests for task package

seelengxd committed:
Merge pull request #9 from seelengxd/branch-A-MoreTesting

A-MoreTesting
